### PR TITLE
Added ECR gate

### PR DESCRIPTION
**Pull request by kbarkhqs**

### CODE REVIEW

File: qoqo/src/operations/single_qubit_gate_operations.rs [hunk 1]
```diff
@@ -505,6 +505,12 @@ struct GPi2 {
 )]
 /// The Identity gate.
 ///
+/// .. math::
+///     U = \frac{1}{\sqrt{2}} \begin{pmatrix}
```
Comment (mlodi-hqs) [inline qoqo/src/operations/single_qubit_gate_operations.rs:509]: The fraction is incorrect, right?

File: qoqo/tests/integration/operations/single_qubit_gate_operations.rs [hunk 2]
```diff
@@ -2776,6 +2776,7 @@ fn test_pyo3_json_schema(operation: SingleQubitGateOperation) {
         let minimum_version: String = match operation {
             SingleQubitGateOperation::GPi(_) => "1.4.0".to_string(),
             SingleQubitGateOperation::GPi2(_) => "1.4.0".to_string(),
+            SingleQubitGateOperation::Identity(_) => "1.7.0".to_string(),
```
Comment (mlodi-hqs) [inline qoqo/tests/integration/operations/single_qubit_gate_operations.rs:2779]: I'm curious, why didn't this fail before?

File: roqoqo/src/operations/two_qubit_gate_operations.rs [hunk 3]
```diff
@@ -2509,3 +2509,108 @@ impl OperateTwoQubitGate for ControlledRotateXY {
...
+/// Trait for all gate operations acting on exactly two qubits.
+impl OperateTwoQubitGate for EchoCrossResonance {
+    /// Returns [KakDecomposition] of the gate.
+    ///
+    /// # Returns
+    ///
+    /// * struct `KakDecomposition { global_phase, k_vector, circuit_before, circuit_after }`
+    fn kak_decomposition(&self) -> KakDecomposition {
+        let mut circuit_b = Circuit::new();
+        circuit_b += SGate::new(self.control);
```
Comment (mlodi-hqs) [inline roqoqo/src/operations/two_qubit_gate_operations.rs:2596]: Is this "basic" enough to keep it here? Should it be decomposed further? It's equivalent to `RotateZ(pi/2)` plus an additional `pi/4` global phase factor. @nfwvogt

**Review by mlodi-hqs — APPROVED** [review]

File: qoqo/tests/integration/operations/single_qubit_gate_operations.rs (hunk 2 above)
Comment (kbarkhqs) [reply qoqo/tests/integration/operations/single_qubit_gate_operations.rs:2779]: It did, but it's behind a feature gate

File: qoqo/src/operations/single_qubit_gate_operations.rs (hunk 1 above)
Comment (kbarkhqs) [reply qoqo/src/operations/single_qubit_gate_operations.rs:509]: Which part of it?

Comment (kbarkhqs) [reply qoqo/src/operations/single_qubit_gate_operations.rs:509]: Ah yes wrong gate! thank you!

**Pull request merged**